scripts/common/longhorn.sh

```diff
@@ -141,10 +141,21 @@ function longhorn_to_sc_migration() {
 # if PVCs and object store data have both been migrated from longhorn and longhorn is no longer specified in the kURL spec, remove longhorn
 function maybe_cleanup_longhorn() {
     if [ -z "$LONGHORN_VERSION" ]; then
+        if kubectl get ns | grep -q longhorn-system; then
+            logStep "Removing Longhorn ..."
+        fi
+
         if [ "$DID_MIGRATE_LONGHORN_PVCS" == "1" ]; then
             report_addon_start "longhorn-removal" "v1"
             remove_longhorn
             report_addon_success "longhorn-removal" "v1"
+            return
+        fi
+
+        logWarn ""
+        logWarn "ATTENTION: Unable to remove Rook."
+        if [ "$DID_MIGRATE_LONGHORN_PVCS" == "1" ]; then
+            logFail "Storage class migration did not succeed"
         fi
     fi
```

Comment:
This logic... Has issues.

For one, we shouldn't log that we're removing longhorn unless `DID_MIGRATE_LONGHORN_PVCS` is `1`. Not even if the namespace is present.

For another, the 
```
logWarn ""
logWarn "ATTENTION: Unable to remove Rook."
```
lines WILL ALWAYS RUN if `LONGHORN_VERSION` isn't set! And it mentions Rook, not Longhorn!

Finally, 
```
        if [ "$DID_MIGRATE_LONGHORN_PVCS" == "1" ]; then
            logFail "Storage class migration did not succeed"
        fi
```
will NEVER run, as it has the same condition as the `remove_longhorn` call above!

Reply:
```suggestion
    if [ -z "$LONGHORN_VERSION" ]; then
        if [ "$DID_MIGRATE_LONGHORN_PVCS" == "1" ]; then
            logStep "Removing Longhorn"
            report_addon_start "longhorn-removal" "v1"
            remove_longhorn
            report_addon_success "longhorn-removal" "v1"
            return
        fi
    fi
```
At most, you can do this - but I would recommend making sure that `remove_longhorn` does not itself have a logstep.

Reply:
No that does not solve the problem.
We must have the logs and know that;
1- We are trying to remove longhorn
2- We did not remove because did not pass and etc. 


See the code now.